cast/src/lib.rs

```diff
@@ -492,6 +492,49 @@ where
             if to_json { serde_json::to_string(&transaction)? } else { to_table(transaction) };
         Ok(transaction)
     }
+
+    /// ```no_run
+    /// use cast::Cast;
+    /// use ethers_providers::{Provider, Http};
+    /// use std::convert::TryFrom;
+    ///
+    /// # async fn foo() -> eyre::Result<()> {
+    /// let provider = Provider::<Http>::try_from("http://localhost:8545")?;
+    /// let cast = Cast::new(provider);
+    /// let tx_hash = "0xf8d1713ea15a81482958fb7ddf884baee8d3bcc478c5f2f604e008dc788ee4fc";
+    /// let receipt = cast.receipt(tx_hash.to_string(), None, false).await?;
+    /// println!("{}", receipt);
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn receipt(
+        &self,
+        tx_hash: String,
+        field: Option<String>,
+        to_json: bool,
+    ) -> Result<String> {
+        let receipt_result = loop {
+            let receipt_result =
+                self.provider.get_transaction_receipt(H256::from_str(&tx_hash)?).await?;
+
+            match receipt_result {
+                Some(receipt_result) => break receipt_result,
+                None => thread::sleep(time::Duration::from_millis(1000)),
```

Comment:
Could we also add something analogous to seth's `--async` flag, where it instead exits immediately if the receipt isn't found?

Reply:
Good looks, will add that flag